File: 3753-maximum-difference-between-even-and-odd-frequency-i/maximum-difference-between-even-and-odd-frequency-i.java
class Solution {
    public int maxDifference(String s) {
        int a1 = Integer.MIN_VALUE, a2 = Integer.MAX_VALUE;
        int[] freq = new int[26];
        
        for (char c : s.toCharArray()) {
            freq[c - 'a']++;
        }

        for (int f : freq) {
            if (f == 0) continue;
            if (f % 2 == 0) a2 = Math.min(a2, f);
            else a1 = Math.max(a1, f);
        }

        if (a1 == Integer.MIN_VALUE || a2 == Integer.MAX_VALUE) return -1;

        return a1 - a2;
    }
}

File: 3753-maximum-difference-between-even-and-odd-frequency-i/maximum-difference-between-even-and-odd-frequency-i.go
func maxDifference(s string) int {
    freq := make([]int, 26);
    for _, c := range s {
        freq[c - 'a']++;
    }

    a1 := -1 << 31;
    a2 := 1 << 31 - 1;

    for _, f := range freq {
        if f == 0 {
            continue;
        }

        if f % 2 == 0 {
            if f < a2 {
                a2 = f;
            }
        } else {
            if f > a1 {
                a1 = f;
            }
        }
    }

    if a1 == -1 << 31 || a2 == 1 << 31 - 1 {
        return -1;
    }

    return a1 - a2;
}